test/jdk/sun/security/util/misc/CurveGetParameterSpec.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8322133
+ * @summary Make sure getParamterSpec returns std name for EC AlgorithmParameters
+ * @modules java.base/sun.security.util
+ */
+
+import java.security.AlgorithmParameters;
+import java.security.KeyPairGenerator;
+import java.security.spec.ECGenParameterSpec;
+
+public class CurveGetParameterSpec {
+    public static void main(String[] args) throws Exception {
+        var k = KeyPairGenerator.getInstance("EC").generateKeyPair().getPublic();
+        var a = AlgorithmParameters.getInstance("EC");
+        a.init(k.getParams());
+        String name = a.getParameterSpec(ECGenParameterSpec.class).getName();
+        if (!name.equals("secp384r1")) {
```

Comment:
The default curve ("secp384r1") may change in the future. Suggest specifying desired curve name with `KeyPairGenerator.init()`.

Also, in the exception message, can you also print the name that was returned?